### PR TITLE
Add retry on delete stack + send event to datadog

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Add a retry around `deleteStack`.
We should add unit tests to verify we do not break it but I did not manage to run them locally yet. I will address that in a follow up PRs.
In the meantime this should reduce some flakiness we have for now

### CODE REVIEW

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 1]
```diff
@@ -277,11 +275,33 @@ func (sm *StackManager) deleteStack(ctx context.Context, stackID string, stack *
...
-	if err != nil {
-		return err
+	downCount := 0
+	var destroyErr error
+	for {
+		downCount++
+		destroyContext, cancel := context.WithTimeout(ctx, stackDestroyTimeout)
+		_, destroyErr = stack.Destroy(destroyContext, progressStreamsDestroyOption, optdestroy.DebugLogging(loggingOptions))
+		cancel()
+		if err == nil {
```
Comment (FlorentClarret) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:290]: Shouldn't this be `destroyErr`?

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/pkg/utils/infra/stack_manager.go (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/pkg/utils/infra/stack_manager.go:290]: Good catch!

File: test/new-e2e/pkg/utils/infra/stack_manager.go [hunk 2]
```diff
@@ -277,11 +275,33 @@ func (sm *StackManager) deleteStack(ctx context.Context, stackID string, stack *
...
+			sendEventToDatadog(ddEventSender, fmt.Sprintf("[E2E] Stack %s : success on Pulumi stack destroy", stackID), "", []string{"operation:destroy", "result:ok", fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", downCount)})
+			break
+		}
+		sendEventToDatadog(ddEventSender, fmt.Sprintf("[E2E] Stack %s : error on Pulumi stack destroy", stackID), destroyErr.Error(), []string{"operation:destroy", "result:fail", fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", downCount)})
+
+		if downCount > stackUpMaxRetry {
+			fmt.Printf("Giving up on error during stack destroy: %v\n", destroyErr)
+			break
```
Comment (FlorentClarret) [inline test/new-e2e/pkg/utils/infra/stack_manager.go:298]: nit - you could return `destroyErr` directly here and remove the if statement after the for loop

Comment (KevinFairise2) [reply test/new-e2e/pkg/utils/infra/stack_manager.go:298]: Indeed, thanks!

**Pull request merged**